receiver/valkeyreceiver/client.go

```diff
@@ -0,0 +1,69 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+package valkeyreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/valkeyreceiver"
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/valkey-io/valkey-go"
+)
+
+var attrDelimiter = "\r\n"
+
+// Interface for a Valkey client. Implementation can be faked for testing.
+type client interface {
+	// retrieves a string of key/value pairs of valkey metadata
+	retrieveInfo(context.Context) (map[string]string, error)
+	// close release Valkey client connection pool
+	close() error
+}
+
+// Wraps a real Valkey client, implements `client` interface.
+type valkeyClient struct {
+	client valkey.Client
+}
+
+var _ client = (*valkeyClient)(nil)
+
+// Creates a new real Valkey client from the passed-in valkey.Options.
+func newValkeyClient(options valkey.ClientOption) (client, error) {
+	innerClient, err := valkey.NewClient(options)
+	if err != nil {
+		return nil, err
+	}
+	return &valkeyClient{innerClient}, nil
+}
+
+// Retrieve Valkey INFO. We retrieve all of the 'sections'.
+func (c *valkeyClient) retrieveInfo(ctx context.Context) (map[string]string, error) {
+	str, err := c.client.Do(ctx, c.client.B().Info().Build()).ToString()
+	if err != nil {
+		return nil, err
+	}
+
+	return parseRawDataMap(str)
+}
+
+func parseRawDataMap(data string) (map[string]string, error) {
+	attrs := make(map[string]string)
+	lines := strings.Split(data, attrDelimiter)
+	for _, line := range lines {
+		if len(line) == 0 || strings.HasPrefix(line, "#") {
```

Comment:
The lines with a `#` prefix appear to start a named section of key/value pairs. Is there a promise in the API that there will be no name conflicts across categories? If no, we may need to consider grouping the parsed pairs by category, not in one shared map.

Reply:
Since `1.0.0`, there do not appear to have been any conflicts in the sections. See the documentation: https://valkey.io/commands/info/

Although I like the idea of grouping the data by section, I would suggest keeping this feature for a future PR if needed. The current exported client metrics are not affected by the tested version (integration test 7.2)." Wdyt?